### PR TITLE
feature/TECH 652 fix spark deploy

### PR DESCRIPTION
**Pull request by koeves**
- [TECH-652] make sparkjob overriden
- [TECH-652] Pass metadata to spark job construction

[TECH-652]: https://vandebron.atlassian.net/browse/TECH-652?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TECH-652]: https://vandebron.atlassian.net/browse/TECH-652?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

----
### 📕 [TECH-652](https://vandebron.atlassian.net/browse/TECH-652) Fix spark job deploy danielkoves@vandebron.nl 

🏗️ Build [2](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-302/2/display/redirect) ✅ Successful, started by _Daniel Koves_  
🚀 *[cloudfront-service](https://cloudfront-service-302.test.nl/swagger/index.html)*, *example-dagster-user-code*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-302.test.nl/swagger/index.html)*, *ocpp-first*, *ocpp-second*, *[sbtservice](https://sbtservice-302.test.nl/swagger/index.html)*, *sparkJob-first*, *sparkJob-second*

### CODE REVIEW

File: src/mpyl/steps/deploy/k8s/resources/spark.py [hunk 1]
```diff
@@ -111,20 +111,20 @@ def get_spark_config_map_data() -> dict:
...
 
 class V1SparkApplication(CustomResourceDefinition):
-    def __init__(self, schedule: Optional[str], body: dict):
+    def __init__(self, metadata: V1ObjectMeta, schedule: Optional[str], body: dict):
         if schedule:
             super().__init__(
                 api_version="sparkoperator.k8s.io/v1beta2",
                 kind="ScheduledSparkApplication",
-                metadata=V1ObjectMeta(name="sparkapplications.sparkoperator.k8s.io"),
+                metadata=metadata,
```
Comment (koeves) [inline src/mpyl/steps/deploy/k8s/resources/spark.py:119]: This was the missing piece btw

File: tests/projects/spark-job/deployment/project.yml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 name: 'sparkJob'
 description: 'Job to enrich charge session with total amount earned'
 maintainer: ['MPyL']
-mpylVersion: 1.0.9
+mpylVersion: 1.4.0
 stages:
   build: Sbt Build
   test: Sbt Test
-  deploy: Kubernetes Spark Job Deploy
```
Comment (Jorg88) [inline tests/projects/spark-job/deployment/project.yml:8]: uhm?

Comment (koeves) [reply tests/projects/spark-job/deployment/project.yml:8]: I made it an override project. @nik-kashi's logic requires you to set the deployment stage in the overrides not in the main one

Comment (Jorg88) [reply tests/projects/spark-job/deployment/project.yml:8]: How do you make something a 'overrride project'? I don't see any difference, guess you made it override compatible? Anyway, if we never need these projects to not be 'override projects' I guess it's fine

**Review by Jorg88 — APPROVED** [review]

File: tests/projects/spark-job/deployment/project.yml (hunk 2 above)
Comment (koeves) [reply tests/projects/spark-job/deployment/project.yml:8]: I added these files: https://github.com/Vandebron/mpyl/pull/302/files#diff-cdf26c82e3b16f52bf356fb05bdae1771f035ffaba4de3a9b973e6e4dfb6c1ce next to the project.yml

**Review by Jorg88 — COMMENTED** [review]
Would be nice to include this in 1.4.4 maybe? Add a release note in that case

**Comment by koeves** [comment]
> Would be nice to include this in 1.4.4 maybe? Add a release note in that case

It's not done yet, Oliver reported issues with the spark deploy. Should be next release I think

**Review by Jorg88 — APPROVED** [review]

**Pull request merged**